frontend/src/components/ibutsu-header.js

```diff
@@ -122,10 +114,33 @@ export class IbutsuHeader extends React.Component {
     if (this.state.filterValue) {
       params['filter'] = ['title%' + this.state.filterValue];
     }
+
+    const {setPrimaryObject, primaryId} = this.context;
+
     // adding s here seems risky, but this scope is small, it's only abstracted for 2 things
     HttpClient.get([Settings.serverUrl, endpoint], params)
       .then(response => HttpClient.handleResponse(response))
-      .then(data => this.setState({projects: data[endpoint+'s'], filteredProjects: data[endpoint+'s']}));
+      .then(data => {
+        // populate context and current input value if passed
+        for ( let item of data[endpoint+'s'] ) {
+          if (item.id === primaryId) {
```

Comment:
is this likely to be one value? 
as seeing a setState in a loop hurts. 

I sugggest 

 const itemYourAfter =  Object.keys(data[endpoint+'s']).find(key=>data[endpoint+'s'][key].id === primaryId)
 
 ...whatevs

Reply:
data[endpoint+'s']?.[key]?.id   << this may be necessary

Reply:
got it! I can see how that's way better.

Reply:
```
        const paramMatch = Object.keys(data[endpoint+'s']).find(key=>data[endpoint+'s'][key].id === primaryObject.id)
        setPrimaryObject(paramMatch || null);
        this.setState(
          {
            projects: param_match?.title || data[endpoint+'s'],
            filteredProjects: param_match ? '' : data[endpoint+'s'],

          })
      }
```

Got this down to the above, thanks for the tip about using `find` here instead of looping. It also prompted me to use primaryObject instead of the param, which needed to change since the param should be synced once to the context.